src/lib/dataEntities/networks.ts

```diff
@@ -282,6 +283,42 @@ export const getL2Network = (
   return getNetwork(signerOrProviderOrChainID, 2) as Promise<L2Network>
 }
 
+/**
+ * Returns the addresses of all contracts that make up the ETH bridge
+ * @param rollupContractAddress Address of the Rollup contract
+ * @param l1SignerOrProvider An L1 signer or provider
+ * @returns EthBridge object with all information about the ETH bridge
+ */
+export const getEthBridgeInformation = async (
+  rollupContractAddress: string,
+  l1SignerOrProvider: SignerOrProvider
+): Promise<EthBridge> => {
+  const rollup = RollupAdminLogic__factory.connect(
+    rollupContractAddress,
+    l1SignerOrProvider
+  )
+
+  const [
+    bridgeContractAddress,
+    inboxContractAddress,
+    outboxContractAddress,
+    sequencerInboxContractAddress,
+  ] = await Promise.all([
+    await rollup.bridge(),
+    await rollup.inbox(),
+    await rollup.outbox(),
+    await rollup.sequencerInbox(),
+  ])
+
+  return {
+    bridge: bridgeContractAddress,
+    inbox: inboxContractAddress,
+    sequencerInbox: sequencerInboxContractAddress,
+    outbox: outboxContractAddress,
+    rollup: rollupContractAddress,
+  }
+}
```

Comment:
Couple of things:
* I think we don't need `await` before the contract calls, as they will return promises, so we get an array of promises that we can await with `Promise.all`
* Nitpick: If we name the properties we destructure from the array the same as the object properties, we can use the shorthand object thing

```suggestion
  const [
    bridge,
    inbox,
    outbox,
    sequencerInbox,
  ] = await Promise.all([
    rollup.bridge(),
    rollup.inbox(),
    rollup.outbox(),
    rollup.sequencerInbox(),
  ])

  return {
    bridge,
    inbox,
    sequencerInbox,
    outbox,
    rollup: rollupContractAddress,
  }
}
```